### PR TITLE
fix handling of structs in imported modules

### LINKED ISSUE #2359: Can't import interface using structs
### Version Information

* vyper Version (output of `vyper --version`): 0.2.12+commit.2c6842c
* OS: linux
* Python Version (output of `python --version`): 3.8.5
* Environment (output of `pip freeze`):
```
asttokens==2.0.4
pycryptodome==3.10.1
semantic-version==2.8.5
six==1.15.0
vyper==0.2.12
```

### What's your issue about?

Can't import an interface if it uses structs. Simple example:

foo.vy:
```
struct Widget:
        name: String[8]
        count: uint256

widget: Widget

@external
def show() -> (String[8], uint256):
        return (self.widget.name, self.widget.count)

@external
def __init__():
        self.widget = Widget({
                name: "thing",
                count: 1
        })
```
bar.vy
```
import foo as Foo

@external
def __init__():
        pass
```

Throw both in the same dir.

`vyper foo.vy` results in a successful compilation

`vyper bar.vy` results in:
```
Error compiling: bar.vy
vyper.exceptions.InvalidType: Invalid base type: Widget
  contract "Foo", line 5:8 
       4
  ---> 5 widget: Widget
  ---------------^
       6
```

### How can it be fixed?

Haven't spent time fixing yet

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
expand parsing to all AST types which might need to be parsed to
semantically understand the top-level items in a module.

Fixes #2359

### How to verify it
Compile the example in the issue

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/ast/signatures/interface.py [hunk 1]
```diff
@@ -99,8 +99,20 @@ def extract_sigs(sig_code, interface_name=None):
...
             i
             for i in vy_ast.parse_to_ast(sig_code["code"], contract_name=interface_name)
-            if isinstance(i, vy_ast.FunctionDef)
-            or isinstance(i, vy_ast.EventDef)
+            # all the nodes visited by ModuleNodeVisitor.
+            if isinstance(
+                i,
+                (
+                    vy_ast.FunctionDef,
+                    vy_ast.EventDef,
```
Comment (fubuloubu) [inline vyper/ast/signatures/interface.py:107]: I'm not sure if Events should be importable, typically they are not interface-able, but I could be convinced (e.g. `Approval` for `ERC20`)

Comment (charles-cooper) [reply vyper/ast/signatures/interface.py:107]: I don't think this really means the Event is importable, it just forces the compiler to register them. I think it's good for overall correctness (imagine if in the future a function signature somehow depends on an Event being present, without parsing Events in this routine there will be a bug in that future).

Comment (fubuloubu) [reply vyper/ast/signatures/interface.py:107]: I suppose this holds for `implements: ERC20`

**Comment by charles-cooper** [comment]
I kind of think struct definitions should actually be importable

**Comment by charles-cooper** [comment]
> I kind of think struct definitions should actually be importable

but maybe this should be a VIP

**Comment by fubuloubu** [comment]
> I kind of think struct definitions should actually be importable

Totally agree

> > I kind of think struct definitions should actually be importable
> 
> but maybe this should be a VIP

Maybe, not sure. I still think it should be in the ABI

File: vyper/ast/signatures/interface.py (hunk 1 above)
Comment (iamdefinitelyahuman) [reply vyper/ast/signatures/interface.py:107]: yes, checking correctness of an interface re: events is why it was added i believe

**Review by iamdefinitelyahuman — APPROVED** [review]

**Pull request merged**